src/drivers/azure-storage-table.ts

```diff
@@ -0,0 +1,143 @@
+import { defineDriver } from "./utils";
+import {
+  TableClient,
+  AzureNamedKeyCredential,
+  AzureSASCredential,
+  TableEntity,
+} from "@azure/data-tables";
+import { DefaultAzureCredential } from "@azure/identity";
+
+export interface AzureStorageTableOptions {
+  /**
+   * The name of the Azure Storage account.
+   * @default null
+   */
+  accountName?: string;
+  /**
+   * The name of the table. All entities will be stored in the same table.
+   * @default 'unstorage'
+   */
+  tableName?: string;
+  /**
+   * The partition key. All entities will be stored in the same partition.
+   * @default 'unstorage'
+   */
+  partitionKey?: string;
+  /**
+   * The account key. If provided, the SAS key will be ignored. Only available in Node.js runtime.
+   * @default null
+   */
+  accountKey?: string;
+  /**
+   * The SAS key. If provided, the account key will be ignored.
+   * @default null
+   */
+  sasKey?: string;
+  /**
+   * The connection string. If provided, the account key and SAS key will be ignored. Only available in Node.js runtime.
+   * @default null
+   */
+  connectionString?: string;
+}
+
+export default defineDriver((opts: AzureStorageTableOptions = {}) => {
+  const {
+    accountName = null,
+    tableName = "unstorage",
+    partitionKey = "unstorage",
+    accountKey = null,
+    sasKey = null,
+    connectionString = null,
+  } = opts;
+  if (!accountName)
+    throw new Error(
+      "Account name is required to use the Azure Storage Table driver."
+    );
+
+  let client: TableClient;
+  if (accountKey) {
+    // AzureNamedKeyCredential is only available in Node.js runtime, not in browsers
+    const credential = new AzureNamedKeyCredential(accountName, accountKey);
+    client = new TableClient(
+      `https://${accountName}.table.core.windows.net`,
+      tableName,
+      credential
+    );
+  } else if (sasKey) {
+    const credential = new AzureSASCredential(sasKey);
+    client = new TableClient(
+      `https://${accountName}.table.core.windows.net`,
+      tableName,
+      credential
+    );
+  } else if (connectionString) {
+    // fromConnectionString is only available in Node.js runtime, not in browsers
+    client = TableClient.fromConnectionString(connectionString, tableName);
+  } else {
+    const credential = new DefaultAzureCredential();
+    client = new TableClient(
+      `https://${accountName}.table.core.windows.net`,
+      tableName,
+      credential
+    );
+  }
```

Comment:
I think we can lazy init `client`. See planetscale as an example.

Reply:
Lazy init of the client would lead to a non shared instance if used in a serverless environment that's why it's initialized at driver creation. Also, the client does not actually connect to Azure, most of the getXClient functions are not doing any network requests, so it's mostly lazy by default.

Reply:
I know. It is for behavior consistency for other drivers (also prevents extra sync overhead and errors on server startup, delaying anything to the first usage from driver)

Reply:
Makes sense, I will make it lazy to have a consistent implementation.